exercises/concept/take-a-number-deluxe/test/take_a_number_deluxe_test.exs

```diff
@@ -0,0 +1,249 @@
+
+      assert TakeANumberDeluxe.report_state(pid) ==
+               %TakeANumberDeluxe.State{
+                 min_number: 1,
+                 max_number: 99,
+                 queue: :queue.in(3, :queue.in(2, :queue.in(1, :queue.new())))
+               }
+    end
+  end
+
+  describe "serve_next_queued_number/1" do
+    @tag task_id: 4
+    test "returns the new number being served" do
+      {:ok, pid} = TakeANumberDeluxe.start_link(min_number: 100, max_number: 999)
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 100}
+      assert TakeANumberDeluxe.serve_next_queued_number(pid) == {:ok, 100}
+    end
+
+    @tag task_id: 4
+    test "can serve multiple numbers" do
+      {:ok, pid} = TakeANumberDeluxe.start_link(min_number: 1, max_number: 99)
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 1}
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 2}
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 3}
+      assert TakeANumberDeluxe.serve_next_queued_number(pid) == {:ok, 1}
+      assert TakeANumberDeluxe.serve_next_queued_number(pid) == {:ok, 2}
+      assert TakeANumberDeluxe.serve_next_queued_number(pid) == {:ok, 3}
+    end
+
+    @tag task_id: 4
+    test "returns an error when there are no more queued" do
+      {:ok, pid} = TakeANumberDeluxe.start_link(min_number: 1, max_number: 3)
+
+      assert TakeANumberDeluxe.serve_next_queued_number(pid) ==
+               {:error, :empty_queue}
+    end
+
+    @tag task_id: 4
+    test "accepts a priority number to skip the queue" do
+      {:ok, pid} = TakeANumberDeluxe.start_link(min_number: 1, max_number: 99)
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 1}
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 2}
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 3}
+      assert TakeANumberDeluxe.serve_next_queued_number(pid, 3) == {:ok, 3}
+    end
+
+    @tag task_id: 4
+    test "returns an error when the priority number is not in the queue" do
+      {:ok, pid} = TakeANumberDeluxe.start_link(min_number: 1, max_number: 3)
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 1}
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 2}
+
+      assert TakeANumberDeluxe.serve_next_queued_number(pid, 7) ==
+               {:error, :priority_number_not_found}
+    end
+
+    @tag task_id: 4
+    test "updates the state accordingly" do
+      {:ok, pid} = TakeANumberDeluxe.start_link(min_number: 1, max_number: 99)
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 1}
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 2}
+      assert TakeANumberDeluxe.queue_new_number(pid) == {:ok, 3}
+      assert TakeANumberDeluxe.serve_next_queued_number(pid) == {:ok, 1}
+      assert TakeANumberDeluxe.serve_next_queued_number(pid) == {:ok, 2}
+
+      queue =
+        :queue.delete(2, :queue.delete(1, :queue.in(3, :queue.in(2, :queue.in(1, :queue.new())))))
```

Comment:
This looks ridiculous, it's not equal to `from_list([3])` and the pipe cannot be used 😂

Reply:
That's one reason why we code in Elixir and not in Erlang :)